models/fivetran_platform__audit_table.sql

```diff
@@ -1,13 +1,13 @@
 {{ config(
-    materialized='table' if is_databricks_sql_warehouse(target) else 'incremental',
+    materialized='incremental' if is_databricks_all_purpose(target) else 'table',
```

Comment:
Just double checking this logic, since the conditions have been flipped:

if `is_databricks_all_purpose(target)` is true, then it'll be materialized as incremental.
If it's false, then it'll be a table. Which makes sense for databricks.

However, what would this entail for the other warehouses? I'm looking at the macro loop and it seems that it would only be true for the case when it's a databricks runtime is all-purpose. But that would be false for the other warehouses, so they would now be materialized as tables instead of incremental. Is that the intention?

Reply:
@fivetran-avinash BRILLIANT catch! You are exactly correct and this was not the intention. This will now set the databricks all-purpose cluster to use the incremental strategy and also turn it off for non all-purpose clusters.... BUT ALSO turns the incremental strategy off for all other warehouses 😱. 

Extremely thankfully you reviewed this and caught this gap. Let me revisit the code and account for all other warehouses.

Reply:
I just made some code updates to account for the above issue. @fivetran-avinash @fivetran-catfritz would you be able to review and let me know if you have any questions or there are any other considerations to take into account?

See below for validations that the materializations are working as expected on each platform.
- ✅  BigQuery (subsequent runs use incremental strategy)
![image](https://github.com/fivetran/dbt_fivetran_log/assets/74217849/3e6f02d4-f930-4e2f-a3f0-4c363b231139)

- ✅  Snowflake (subsequent runs use incremental strategy)
![image](https://github.com/fivetran/dbt_fivetran_log/assets/74217849/b83791cb-ab43-4ba1-b18f-e776b152275c)

- ✅  Redshift (subsequent runs use incremental strategy)
![image](https://github.com/fivetran/dbt_fivetran_log/assets/74217849/82283fc0-2d67-4edb-99fa-e1882e6523b0)

- ✅  Postgres (subsequent runs use incremental strategy)
![image](https://github.com/fivetran/dbt_fivetran_log/assets/74217849/6b6340cd-a948-4909-b43b-a13a03ac499b)

- ✅  Databricks All-Purpose Cluster (subsequent runs use incremental strategy)
![image](https://github.com/fivetran/dbt_fivetran_log/assets/74217849/0702ec22-3fb6-4f7d-9b3a-52a836659bab)

- ✅  Databricks SQL Warehouse (no runs use the incremental strategy)
![image](https://github.com/fivetran/dbt_fivetran_log/assets/74217849/eb7edf2a-4b03-42b2-92d4-dfc3b1e8f2dd)